### PR TITLE
[feature, model]Fix Qwen3-Omni Megatron HF corr checks

### PR DESCRIPTION
**Pull request by hbhflw2000**
## What

This PR tightens Qwen3-Omni Megatron/HF parity support for fixed-score and local-attention validation paths.

- Select the local GPT layer spec when Qwen3-Omni is configured with the local attention backend, while keeping the Transformer Engine spec as the default when TE is available.
- Install the mRoPE-aware `Qwen3VLSelfAttention` module in the Qwen3-Omni language transformer spec.
- Add a local causal attention fallback for padded fixed-score paths that use `DotProductAttention`.
- Update dense context-parallel batch slicing to pass the current `get_batch_on_this_cp_rank(..., is_hybrid_cp=False, ...)` API.
- Avoid view aliasing in the Qwen3-VL deepstack visual update path.
- Add a small no-Ray fixed-score correlation checker for Megatron actor JSON vs. HF JSON.

## Why

Qwen3-Omni Megatron/HF parity depends on two independent pieces being correct:

- MoE EP expert parameters must map rank-local expert indices back to global expert indices before comparing or exporting against HF.
- Attention execution must preserve Qwen3-VL's mRoPE behavior in both the TE-backed path and the local/fallback path.

The EP local-to-global expert mapping is already present in the base used by this branch. This PR focuses on the second part: local-attention/fixed-score paths and a lightweight checker that makes parity regressions easier to catch.

Qwen3-Omni parity checks can exercise local attention and dense CP code paths even when the main training environment uses the regular TE-backed path. The affected paths need to:

- preserve Qwen3-VL's absolute/mRoPE attention behavior,
- match the current Megatron-Core context-parallel helper signature,
- avoid view aliasing when updating visual tokens in-place, and
- provide a lightweight way to validate actor/HF fixed-score correlation without starting the RL stack.

## Compatibility / Risk

The local-attention changes are gated by backend and module type:

- In the default community environment with TE available and a non-local attention backend, Qwen3-Omni still builds from `get_gpt_layer_with_transformer_engine_spec`.
- The local spec is selected only when `attention_backend` is `AttnBackend.local` or `"local"`, or when TE is unavailable.
- The new local causal attention fallback only runs for `DotProductAttention` with no packed-sequence params, no attention bias, and static batching. Other attention implementations continue through the existing core-attention path.

This means an environment that has TE/apex installed should continue using the TE-backed default path. The fallback mainly covers local/fixed-score validation paths and environments where TE is not available.

The EP expert-index issue is separate from TE availability. If the local-to-global EP mapping is missing, actor/HF correlation can still regress even in TE-backed runs. Conversely, this PR's local-attention fallback is intended to avoid additional correlation drops when TE is unavailable or the local backend is selected.

## Testing

```bash
python -m pytest -q tests/unit_tests/tools/test_qwen3_omni_fixed_score_corr.py
```

Result:

```text
4 passed
```

```bash
python -m py_compile \
  tools/qwen3_omni_fixed_score_corr.py \
  tests/unit_tests/tools/test_qwen3_omni_fixed_score_corr.py \
  src/megatron/bridge/models/qwen_omni/qwen3_omni_provider.py \
  src/megatron/bridge/models/qwen_omni/qwen3_omni_step.py \
  src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/attention.py \
  src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/transformer_block.py \
  tests/unit_tests/models/qwen_omni/test_qwen3_omni_provider.py \
  tests/unit_tests/models/qwen_omni/test_qwen3_omni_step.py \
  tests/unit_tests/models/qwen_vl/modelling_qwen3_vl/test_attention.py \
  tests/unit_tests/models/qwen_vl/modelling_qwen3_vl/test_transformer_block.py
```

Result: passed.

Fixed-score artifact validation:

```text
num_common=64
pearson=0.999813913
diff_mean=0.006626420
abs_diff_mean=0.037385486
abs_diff_max=0.439933777
PASS
```

### CODE REVIEW

**Review by yaoyu-33 — DISMISSED** [review]

**Comment by yaoyu-33** [comment]
/ok to test https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4567/commits/10fd9842a8a75bafe5f34fd13c4a62a9b83540d7

**Comment by hbhflw2000** [comment]
@yaoyu-33 Sorry for the noise. I saw the lint failure and tried to fix it, but I should have checked the PR state first.

I’ve reverted the branch back to the previous head `10fd984`. Apologies for disturbing the ready-to-merge state. I won’t push further unless requested.

**Comment by suiyoubi** [comment]
Hi @hbhflw2000 , I would suggest you pull the latest main and fix the lint and copyright issue. And we could re-run the test. Thanks!

**Comment by hbhflw2000** [comment]
Hi @suiyoubi , thanks for the suggestion. I've rebased & fixed the lint issues. Please feel free to re-run the tests.

**Comment by hbhflw2000** [comment]
Hi @liding-nv , I pushed a small test-only update to improve patch coverage(failed in last CI test) for the TE/local attention provider paths. Could you please re-run the CI when convenient? Thx

**Comment by huvunvidia** [comment]
Hi @hbhflw2000 ,
The test "[CICD NeMo / L0_Launch_training_local_checkpointing (push)](https://github.com/NVIDIA-NeMo/Megatron-Bridge/actions/runs/28951790766/job/86037275533?pr=4567)" broke.
Can you have a look?

**Comment by hbhflw2000** [comment]
Hi @huvunvidia  ，Thanks for checking. I double checked the failure, and it does not look related to this PR.

The failing test is `TestLocalCheckpointing.test_local_checkpoint_save_resume_with_most_recent_k`. It fails during the second `pretrain()` call after resuming from a local checkpoint, where `masked_next_token_loss()` detects a NaN local forward loss at iteration 8.

This PR only touches Qwen3-Omni/Qwen3-VL model paths, related unit tests, and the standalone fixed-score correlation helper. It does not modify the local checkpointing, GPT training, loss, optimizer, scheduler, or this functional test path.

One possible hint is that this test exercises the existing local checkpoint resume path with `non_persistent_local_ckpt_algo="atomic"` and `most_recent_k=1`. If it reproduces, I think the local checkpoint resume/retention path would be the more relevant area to inspect.

Could we re-run the failed job to confirm? Thanks very much～

**Comment by huvunvidia** [comment]
/ok to test [0bfbc87](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4567/commits/0bfbc877cfbf5dc6bb84ecf45dfaccdeaff1706c)

**Review by huvunvidia — APPROVED** [review]
Looks good. All tests passed.
